### PR TITLE
Loesung_06 Jan Merz (1778099, 1623571)

### PR DESCRIPTION
**Pull request by TrueEsel**
task 1: 1778099
task 2: 1623571

### CODE REVIEW

File: Exercises/JanMerz/exercises_06.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+#     - j (file, size=4060174)
+#     - d.log (file, size=8033020)
+#     - d.ext (file, size=5626152)
+#     - k (file, size=7214296)
+#
+#
+# PART 1:
+# Write a function that reads a console session like the example above from a file, 
+# and then returns a dictionary listing the size of each directory. Directories should
+# be identified by their full path. For the example above:
```
Comment (DrKeks1010) [inline Exercises/JanMerz/exercises_06.py:64]: Sadly, you didn't write a function compliting this task. You did although write code to produce the required dictionary. You should therefor move the code in a seperate funcion, which will require the console session as an input parameter. Overall this should be a quick fix, as you already wrote all the necessary code.

File: Exercises/JanMerz/exercises_06.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        self.name = name
+        self.size = size
+        #self.contents = contents
+
+    def add_content(self, directory_dictionary: dict, name: str, size: int = 0):
+        if(size == 0):
+            directory_dictionary[name] = Directory(name)
+        else:
+            self.change_own_size(size, directory_dictionary)
+        return directory_dictionary
```
Comment (DrKeks1010) [inline Exercises/JanMerz/exercises_06.py:95]: -- not mandatory --
returning the directory is unnecessary as the change of properties of objects in a dictionary will be done in place. So you might remove this line to improve readability.

File: Exercises/JanMerz/exercises_06.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        if(line.startswith("$ cd")):
+            if(line.endswith("..")):
+                path = path[0:path.rfind("\\")]
+            else:
+                old_path = path
+                #substracts the cd command and only leaves the dir name, which is added to the path.
+                #Example: $ cd bwmglvmt is bwmglvmt, which is added to the path: /\pcqjnl\lrrl\nwjggvr\bwmglvmt
+                path += "\\" + re.sub(r'^(.*?)\$ cd ', '', line)
+                if(path not in directory_dictionary):
+                    directory_dictionary = directory_dictionary[old_path].add_content(directory_dictionary, path)
```
Comment (DrKeks1010) [inline Exercises/JanMerz/exercises_06.py:149]: -- not mandatory --
with the change suggestet above, you should also then remove the assignment here

File: Exercises/JanMerz/exercises_06.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+        if(in_ls_output):
+            if(line[0].isdigit()):
+
+                #matches the number (file size) of a file. Example: 68377 jvdqjhr.jvp is 68377
+                file_size = int(re.match(r'\d*', line).group())
+                #substracts everything thats the number (file size) and following white space and adds the remainin string to a path.
+                #Example: 68377 jvdqjhr.jvp is /\jvdqjhr.jvp
+                file_name = path + "\\" + re.sub(r'^\S* ', '', line)
+                directory_dictionary = directory_dictionary[path].add_content(directory_dictionary, file_name, file_size)
```
Comment (DrKeks1010) [inline Exercises/JanMerz/exercises_06.py:131]: -- not mandatory --
with the change suggestet above, you should also then remove the assignment here

**Review by DrKeks1010 — CHANGES_REQUESTED** [review]
Great solution overall. But you missed a small part of the assignment.

File: Exercises/JanMerz/exercises_06.py (hunk 1 above)
Comment (TrueEsel) [reply Exercises/JanMerz/exercises_06.py:64]: solved this problem by making build_directory return the necessary dicitonary

File: Exercises/JanMerz/exercises_06.py (hunk 2 above)
Comment (TrueEsel) [reply Exercises/JanMerz/exercises_06.py:95]: removed return

File: Exercises/JanMerz/exercises_06.py (hunk 3 above)
Comment (TrueEsel) [reply Exercises/JanMerz/exercises_06.py:149]: removed assignment

File: Exercises/JanMerz/exercises_06.py (hunk 4 above)
Comment (TrueEsel) [reply Exercises/JanMerz/exercises_06.py:131]: removed assignment

**Review by DrKeks1010 — APPROVED** [review]
Great Job

**Comment by CorvinAn97** [comment]
Overall a great Solution

**Pull request merged**